nibabel/nifti1.py

```diff
@@ -698,16 +698,26 @@ def get_data_shape(self):
         Allows for freesurfer hack for large vectors described in
         https://github.com/nipy/nibabel/issues/100 and
         http://code.google.com/p/fieldtrip/source/browse/trunk/external/freesurfer/save_nifti.m?spec=svn5022&r=5022#77
+
+        Allows for freesurfer hack for 7th order icosahedron surface described
+        in
+        https://github.com/nipy/nibabel/issues/309
+        https://code.google.com/p/fieldtrip/source/browse/trunk/external/freesurfer/load_nifti.m?r=8776#86
+        https://code.google.com/p/fieldtrip/source/browse/trunk/external/freesurfer/save_nifti.m?r=8776#50
         '''
         shape = super(Nifti1Header, self).get_data_shape()
         # Apply freesurfer hack for vector
-        if shape != (-1, 1, 1): # Normal case
+        if shape == (-1, 1, 1):
+            vec_len = int(self._structarr['glmin'])
+            if vec_len == 0:
+                raise HeaderDataError('-1 in dim[1] but 0 in glmin; '
+                                      'inconsistent freesurfer type header?')
+            return (vec_len, 1, 1)
+        # Apply freesurfer hack for ico7 surface
+        elif shape == (27307, 1, 6):
```

Comment:
To surfaces with that number of vertices. I can't find a good reference to the arguments as to why they do this[0], but a reasonably common FreeSurfer surface is a warped icosahedron, and in particular they [recommend](https://surfer.nmr.mgh.harvard.edu/fswiki/mri_surf2surf#line-31) using the 7th-order icosahedron. This is the number of vertices used in their `fsaverage` group template for comparisons across subjects.

Their MATLAB code indicates that this specific surface shape is hard-coded:
https://code.google.com/p/fieldtrip/source/browse/trunk/external/freesurfer/save_nifti.m?r=8776#50
https://code.google.com/p/fieldtrip/source/browse/trunk/external/freesurfer/load_nifti.m?r=8776#86

[0] I believe the use of the icosahedron is to allow for near constant neighbor-neighbor distances.